src/view/writer.js

```diff
@@ -12,704 +12,993 @@ import ContainerElement from './containerelement';
+	 *
+	 *		// Sets ranges from the iterable of ranges.
+	 * 		const ranges = [ new Range( start1, end2 ), new Range( star2, end2 ) ];
+	 *		writer.setSelection( range, isBackwardSelection );
+	 *
+	 *		// Sets ranges from the other selection.
+	 *		const otherSelection = new Selection();
+	 *		writer.setSelection( otherSelection );
+	 *
+	 * 		// Sets collapsed range at the given position.
+	 *		const position = new Position( root, path );
+	 *		writer.setSelection( position );
+	 *
+	 * 		// Sets collapsed range on the given item.
+	 *		const paragraph = writer.createElement( 'paragraph' );
+	 *		writer.setSelection( paragraph, offset );
+	 *
+	 * 		// Removes all ranges.
+	 *		writer.setSelection( null );
+	 *
+	 * @param {module:engine/view/selection~Selection|module:engine/view/position~Position|
+	 * Iterable.<module:engine/view/range~Range>|module:engine/view/range~Range|module:engine/view/item~Item|null} selectable
+	 * @param {Boolean|Number|'before'|'end'|'after'} [backwardSelectionOrOffset]
+	 */
+	setSelection( selectable, backwardSelectionOrOffset ) {
+		this.document.selection._setTo( selectable, backwardSelectionOrOffset );
 	}
 
-	if ( !element.parent ) {
-		/**
-		 * Trying to break root element.
-		 *
-		 * @error view-writer-break-root
-		 */
-		throw new CKEditorError( 'view-writer-break-root: Trying to break root element.' );
+	/**
+	 * Moves {@link module:engine/view/selection~Selection#focus selection's focus} to the specified location.
+	 *
+	 * The location can be specified in the same form as {@link module:engine/view/position~Position.createAt} parameters.
+	 *
+	 * @param {module:engine/view/item~Item|module:engine/view/position~Position} itemOrPosition
+	 * @param {Number|'end'|'before'|'after'} [offset=0] Offset or one of the flags. Used only when
+	 * first parameter is a {@link module:engine/view/item~Item view item}.
+	 */
+	setSelectionFocus( itemOrPosition, offset ) {
+		this.document.selection._setFocus( itemOrPosition, offset );
 	}
 
-	if ( position.isAtStart ) {
-		return Position.createBefore( element );
-	} else if ( !position.isAtEnd ) {
-		const newElement = element.clone( false );
+	/**
+	 * Sets {@link module:engine/view/selection~Selection selection's} to be marked as `fake`. A fake selection does
+	 * not render as browser native selection over selected elements and is hidden to the user.
+	 * This way, no native selection UI artifacts are displayed to the user and selection over elements can be
+	 * represented in other way, for example by applying proper CSS class.
+	 *
+	 * Additionally fake's selection label can be provided. It will be used to describe fake selection in DOM (and be
+	 * properly handled by screen readers).
+	 *
+	 * @param {Boolean} [value=true] If set to true selection will be marked as `fake`.
+	 * @param {Object} [options] Additional options.
+	 * @param {String} [options.label=''] Fake selection label.
+	 */
+	setFakeSelection( value, options ) {
```

Comment:
I don't like this method name. Since we use `setSelection` to set ranges I would expect that this method does something with ranges. The better name would be `fakeSelection`. However,  it is still not perfect. You chnage selection direction using `setSelection`, but need to use the separate method to make it fake. We could change `backwardSelection` parameter of the selection to the `option` parameter and use it also to set fake selections. I will create a follow-up about it.